### DISCUSSION TITLE
Switch from cluster mode to namespaced mode?

### DISCUSSION THREAD

**Discussion by piggywave** [argoproj/argo-workflows#12250] [Q&A] [answered]
# Background
We have a older version cluster wide instance and a lastest namespaced instance in a same k8s cluster.  The configuration of workflow in namespaced instance will be overridden by cluster instance (eg: `--executor-image`). So we are seeking to change cluster instance to managed-namespaces smoothly.

I did not find any doc about switch  installation mode, remind me if there is any. Thanks!

## what I have done
I run the older cluster version in kind instance, then change the manifests with `--namespaced` and `--managed-namespace` flags.
workflows work well in specified namespaces, but will be pending in others (workflow created without creating pod)

**Comment by agilgur5** [comment, marked as answer]
> I did not find any doc about switch installation mode, remind me if there is any. Thanks!

There is no formal process for that. You would just delete your existing Controller and start a new one with the changed mode.

Changing the manifests, as you said you did, and applying them would result in an equivalent change; the Controller Deployment would be restarted with the new configuration.

> workflows work well in specified namesapces, but will be pending in others(workflow created without creating pod)

Yes, that would be expected, given that your Controller is no longer watching those namespaces.
